### PR TITLE
feat(rds): add support for monitoring to DatabaseCluster

### LINKED ISSUE #2826: Add support for monitoring to RDS DatabaseCluster
At the moment is not possible to specify any monitoring configuration on DatabaseCluster instances. This kind of feature would improve ops capabilities. Ideally it have to be possible to specify a `monitoringInterval` and auto-generate a valid _Role_ to manage it.

### PR DESCRIPTION
**Pull request by rpanfili**
Lets a user specify a monitoring interval period. This change will also auto-generate a valid Role to manage DB instances monitoring.

This change adds an optional prop `monitoringInterval`

closes #2826

BREAKING CHANGE:
    * rds: `monitoringInterval` in `DatabaseInstanceNewProps` has been renamed to `monitoringIntervalSec`

----

### Pull Request Checklist

* [x] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [x] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
  - __Design__: For significant features, design document added to `design` folder
* [x] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** and module name in parens, which will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/lib/cluster.ts [hunk 1]
```diff
@@ -128,6 +129,14 @@ export interface DatabaseClusterProps {
...
    */
   readonly deleteReplacePolicy?: DeletionPolicy
+
+  /**
+   * The interval, in seconds, between points when Amazon RDS collects enhanced
+   * monitoring metrics for the DB instances.
+   *
+   * @default no enhanced monitoring
+   */
+  readonly monitoringInterval?: number;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-rds/lib/cluster.ts]: Please annotate this property with a unit. Recommended: `monitoringIntervalSec`.

Although @RomainMuller is currently working on a `Duration` class which will be the recommended practice going forward.

**Comment by rix0rrr** [comment]
Other than that, looks good!

File: packages/@aws-cdk/aws-rds/lib/cluster.ts (hunk 1 above)
Comment (jogold) [reply packages/@aws-cdk/aws-rds/lib/cluster.ts]: could be also adapted in `DatabaseInstance` then.

https://github.com/awslabs/aws-cdk/blob/e0718ef7cc26798ec2401451be9d7b7894f0b8bc/packages/%40aws-cdk/aws-rds/lib/instance.ts#L375

(breaking)

Comment (rpanfili) [reply packages/@aws-cdk/aws-rds/lib/cluster.ts]: I renamed that property in `DatabaseClusterProps` and in `DatabaseInstanceNewProps` as suggested.
I also added the _Breaking_ advise and rebased to _master_!

Comment (rpanfili) [reply packages/@aws-cdk/aws-rds/lib/cluster.ts]: @rix0rrr I've seen that the `Duration` class has landed on master branch so I updated this PR with those required changes. Let me know if you need anything else!

**Comment by rix0rrr** [comment]
Needs a merge from master, unfortunately :(

**Comment by rpanfili** [comment]
@rix0rrr rebased!

**Comment by rpanfili** [comment]
> @rix0rrr rebased!

and rebased again :)

File: packages/@aws-cdk/aws-rds/lib/cluster.ts [hunk 2]
```diff
@@ -345,6 +354,17 @@ export class DatabaseCluster extends DatabaseClusterBase {
 
     // Get the actual subnet objects so we can depend on internet connectivity.
     const internetConnected = props.instanceProps.vpc.selectSubnets(props.instanceProps.vpcSubnets).internetConnectivityEstablished;
+
+    let monitoringRole;
+    if (props.monitoringInterval && props.monitoringInterval.toSeconds()) {
+      monitoringRole = new Role(this, "MonitoringRole", {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/cluster.ts]: It's nice that we default this, but can we also allow specifying this in `props`? `monitoringRole` is probably a good name for that property.

File: packages/@aws-cdk/aws-rds/lib/cluster.ts [hunk 3]
```diff
@@ -345,6 +361,17 @@ export class DatabaseCluster extends DatabaseClusterBase {
 
     // Get the actual subnet objects so we can depend on internet connectivity.
     const internetConnected = props.instanceProps.vpc.selectSubnets(props.instanceProps.vpcSubnets).internetConnectivityEstablished;
+
+    let monitoringRole;
+    if (props.monitoringInterval && props.monitoringInterval.toSeconds()) {
+      monitoringRole = props.monitoringRole || new Role(this, "MonitoringRole", {
```
Comment (jogold) [inline packages/@aws-cdk/aws-rds/lib/cluster.ts]: It would be nice to have the same optional `monitoringRole` for an instance.

Comment (rpanfili) [reply packages/@aws-cdk/aws-rds/lib/cluster.ts]: ok! done

File: packages/@aws-cdk/aws-rds/lib/cluster.ts (hunk 2 above)
Comment (rpanfili) [reply packages/@aws-cdk/aws-rds/lib/cluster.ts]: ok, great!

**Pull request merged**